src/unix/uiactionx11.cpp

```diff
@@ -46,6 +55,113 @@ GtkWidget* wxGetTopLevelGTK();
 
 namespace
 {
+// Like the real events, this class tries to add a _fake_ delay to the generated
+// (fake) events so that chances that they got lost or ignored (for whatever reasons)
+// by the X server (or the WM, or even the input device driver) are minimized.
+class wxXSync
+{
+public:
+    wxXSync(wxX11Display& display)
+        : m_display(display), m_isMotion(true)
+    {
+    }
+
+    wxXSync(wxX11Display& display, bool depressed)
+        : m_display(display), m_isMotion(false)
+    {
+        static int s_maxDepressed = 0;
+
+        if ( depressed )
+        {
+            ++ms_numDepressed; // A key or button is down.
+
+            if ( ms_numDepressed - s_maxDepressed != 1 )
+                wxFAIL_MSG("Invalid call to wxXSync() ctor");
+
+            s_maxDepressed = ms_numDepressed;
+        }
+        else
+        {
+            if ( s_maxDepressed != 0 )
+            {
+                s_maxDepressed = 0; // reset
+
+                wxYield();
+                wxMilliSleep(Default_Delay);
+            }
+
+            --ms_numDepressed; // A key or button is up.
+
+            if ( ms_numDepressed < 0 )
+                wxFAIL_MSG("Invalid call to wxXSync() ctor");
+        }
+    }
+
+    ~wxXSync()
+    {
+        XSync(m_display, False);
+
+        m_isMotion ? OnMotion() : OnKeyOrButtonUp();
```

Comment:
I find this code slightly confusing, why do we need these `OnXXX()` methods? This naming convention is usually reserved for the event handlers, but they don't handle anything... I'd call them `DoAfterXXX()` or something like that.

But, in fact, looking at their code, I'd probably just inline them here.